src/fides/api/ops/db/base_class.py

```diff
@@ -32,3 +48,276 @@ def process_result_value(self, value: str, _: Any | None) -> dict[str, Any] | No
         if value is not None:
             return json.loads(value)
         return value
+
+
+def get_key_from_data(data: dict[str, Any], cls_name: str) -> str:
+    """Extracts key from data, validates.
+
+    If no key, uses a snake-cased name. Will be used as the URL onupdate
+    applicable classes.
+    """
+    key = FidesKey.validate(data.get("key")) if data.get("key") else None
+    if key is None:
+        name = data.get("name")
+        if name is None:
+            raise KeyValidationError(f"{cls_name} requires a name.")
+        key = FidesKey.validate(to_snake_case(name))
+    return key
+
+
+class FidesBase:
```

Comment:
Oh, that is odd. I'm going to leave it for now since the scope is already large enough here, but as I slowly refactor ctl into ops I'll try to revisit that and clean it up